File: list_executator.py
import datetime


class ListExecutator():
    def __init__(self, service):
        self.service = service

    def get_events(self):
        # Call the Calendar API
        now = datetime.datetime.utcnow().isoformat() + 'Z'  # 'Z' indicates UTC time
        
        print('Getting the upcoming 10 events')
        return self.service.events().list(
            calendarId='primary', timeMin=now,
            maxResults=10, singleEvents=True,
            orderBy='startTime'
        ).execute()

    def execute(self):
        events_result = self.get_events()
        events = events_result.get('items', [])

        if not events:
            return print('No upcoming events found.')

        # Prints the start and name of the next 10 events
        for event in events:
            start = event['start'].get('dateTime', event['start'].get('date'))
            print(start, event.get('summary', event['id']))

File: golang/list_executator.go
package main

import (
	"fmt"
	"log"
	"time"

	"google.golang.org/api/calendar/v3"
)

type ListExecutator struct {
	service *calendar.Service
}

func NewListExecutator(service *calendar.Service) *ListExecutator {
	return &ListExecutator{
		service: service,
	}
}

func (le *ListExecutator) GetEvents() (*calendar.Events, error) {
	now := time.Now().Format(time.RFC3339)

	fmt.Println("Getting the upcoming 10 events")
	events, err := le.service.Events.List("primary").TimeMin(now).MaxResults(10).SingleEvents(true).OrderBy("startTime").Do()
	if err != nil {
		return nil, fmt.Errorf("failed to retrieve events: %v", err)
	}

	return events, nil
}

func (le *ListExecutator) Execute() {
	events, err := le.GetEvents()
	if err != nil {
		log.Fatalf("Failed to get events: %v", err)
	}

	if len(events.Items) == 0 {
		fmt.Println("No upcoming events found.")
		return
	}

	fmt.Println("Upcoming events:")
	for _, event := range events.Items {
		start := event.Start.DateTime
		if start == "" {
			start = event.Start.Date
		}
		fmt.Printf("%s - %s\n", start, event.Summary)
	}
}

func main() {
	listExecutator := NewListExecutator(service)
	listExecutator.Execute()
}
